src/main/java/net/mcreator/ui/modgui/codeviewer/ModElementCodeViewer.java

```diff
@@ -1,126 +1,126 @@
-import javax.swing.*;
-import java.awt.*;
-import java.awt.event.ComponentAdapter;
-import java.awt.event.ComponentEvent;
-import java.io.File;
-import java.util.Comparator;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-public class ModElementCodeViewer<T extends GeneratableElement> extends JTabbedPane {
-
-	private final ModElementGUI<T> modElementGUI;
-
-	private final Map<File, FileCodeViewer<T>> cache = new HashMap<>();
-
-	private boolean updateRunning = false;
-	private final ModElementChangedListener codeChangedListener;
-
-	public ModElementCodeViewer(ModElementGUI<T> modElementGUI) {
-		super(JTabbedPane.BOTTOM);
-
-		this.modElementGUI = modElementGUI;
-		this.codeChangedListener = this::reload;
-
-		setBackground((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"));
-		setTabLayoutPolicy(JTabbedPane.SCROLL_TAB_LAYOUT);
-		setOpaque(true);
-
-		addComponentListener(new ComponentAdapter() {
-			@Override public void componentShown(ComponentEvent e) {
-				super.componentShown(e);
-				reload();
-			}
-		});
-	}
-
-	public void registerUI(JComponent container) {
-		codeChangedListener.registerUI(container, true);
-	}
-
-	private void reload() {
-		if (isVisible() && !updateRunning) {
-			updateRunning = true;
-			new Thread(() -> {
-				try {
-					List<GeneratorFile> files = modElementGUI.getModElement().getGenerator()
-							.generateElement(modElementGUI.getElementFromGUI(), false, false);
-
-					files.sort(
-							Comparator.comparing(e -> FilenameUtils.getExtension(((GeneratorFile) e).file().getName()))
-									.thenComparing(e -> ((GeneratorFile) e).file().getName()));
-
-					for (GeneratorFile file : files) {
-						if (cache.containsKey(file.file())) { // existing file
-							SwingUtilities.invokeAndWait(() -> {
-								try {
-									if (cache.get(file.file()).update(file)) {
-										int tabid = indexOfComponent(cache.get(file.file()));
-										if (tabid != -1)
-											setSelectedIndex(tabid);
-									}
-								} catch (Exception ignored) {
-								}
-							});
-						} else { // new file
-							SwingUtilities.invokeAndWait(() -> {
-								try {
-									FileCodeViewer<T> fileCodeViewer = new FileCodeViewer<>(this, file);
-									addTab(file.file().getName(), FileIcons.getIconForFile(file.file()),
-											fileCodeViewer);
-									cache.put(file.file(), fileCodeViewer);
-								} catch (Exception ignored) {
-								}
-							});
-						}
-					}
-
-					cache.keySet().stream().toList().forEach(file -> {
-						if (!files.stream().map(GeneratorFile::file).toList().contains(file)) { // deleted file
-							remove(cache.get(file));
-							cache.remove(file);
-						}
-					});
-					setBackground((Color) UIManager.get("MCreatorLAF.LIGHT_ACCENT"));
-				} catch (Exception ignored) {
-					setBackground(new Color(0x8D5C5C));
-				}
-				updateRunning = false;
-			}).start();
-		}
-	}
-
-	public ModElementGUI<T> getModElementGUI() {
-		return modElementGUI;
-	}
-
-}
+/*
```

Comment:
Is this line separator chabge OKay? Or I need to revert it?
The same about #2010.

Reply:
Wait, it was CR-LF and changed to LF, I don't like that. I'll find some time to revert it.